scripts/mutect2_wdl/unsupported/mutect2_opt.wdl

```diff
@@ -412,77 +488,152 @@ task Filter {
     fi
 
     # Use GATK Jar override if specified
-    GATK_JAR=${gatk4_jar}
-    if [[ "${gatk4_jar_override}" == *.jar ]]; then
-        GATK_JAR=${gatk4_jar_override}
-    fi
+    GATK_JAR=${default="/root/gatk.jar" gatk4_jar_override}
 
-    touch contamination.table
-    if [[ "${variants_for_contamination}" == *.vcf ]]; then
-        java -Xmx6g -jar $GATK_JAR GetPileupSummaries -I ${tumor_bam} ${"-L " + intervals} -V ${variants_for_contamination} -O pileups.table
-        java -Xmx6g -jar $GATK_JAR CalculateContamination -I pileups.table -O contamination.table
-        contamination_cmd="-contaminationTable contamination.table"
+    java -Xmx${command_mem}m -jar $GATK_JAR GetPileupSummaries -I ${tumor_bam} ${"-L " + intervals} -V ${variants_for_contamination} -O pileups.table
+    java -Xmx${command_mem}m -jar $GATK_JAR CalculateContamination -I pileups.table -O contamination.table
+  }
+
+  runtime {
+    docker: "${m2_docker}"
+    memory: command_mem + " MB"
+    disks: "local-disk " + disk_space_gb + " HDD"
+    preemptible: select_first([preemptible_attempts, 5])
+  }
+
+  output {
+    File contamination_table = "contamination.table"
+  }
+}
+
+task Filter {
+  File? gatk4_jar_override
+  File unfiltered_vcf
+  String output_vcf_name
+  String? m2_extra_filtering_args
+  File? contamination_table
+
+  # Runtime parameters
+  Int preemptible_attempts
```

Comment:
Why does preemptible attempts here (and elsewhere in tasks) need to be required?

Reply:
probably a copy paste error ><